dll/win32/shell32/CFolderOptions.cpp

```diff
@@ -113,6 +113,45 @@ HRESULT STDMETHODCALLTYPE CFolderOptions::GetSite(REFIID riid, void **ppvSite)
 /*************************************************************************
  * FolderOptions helper methods
  */
+static HRESULT ResetGlobalFolderSettings()
+{
+    IGlobalFolderSettings *pgfs;
+    HRESULT hr = CoCreateInstance(CLSID_GlobalFolderSettings, NULL, CLSCTX_INPROC_SERVER,
+                                  IID_PPV_ARG(IGlobalFolderSettings, &pgfs));
+    if (SUCCEEDED(hr))
+    {
+        hr = pgfs->Set(NULL, 0, 0);
+        pgfs->Release();
+    }
+    return hr;
+}
+
+static inline HRESULT ExecResetDefViewFolderSettings(IUnknown *pUnk)
+{
+    return IUnknown_Exec(pUnk, CGID_DefView, DVCMDID_RESET_DEFAULTFOLDER_SETTINGS,
+                         OLECMDEXECOPT_DODEFAULT, NULL, NULL);
+}
+
+static HRESULT ResetDefViewFolderSettings()
+{
+    CComPtr<IShellFolder> pSF;
+    HRESULT hr = SHGetDesktopFolder(&pSF);
+    if (SUCCEEDED(hr))
+    {
+        CComPtr<IShellView> pSV;
+        SFV_CREATE create = { sizeof(SFV_CREATE), pSF };
+        if (SUCCEEDED(hr = SHCreateShellFolderView(&create, &pSV)))
```

Comment:
Does this require a "full" shell-folder-view creation? (I mean, is it not possible to do a QueryInterface or a IUnknown_QueryService call for IShellView on pSF ?)
NOTE: This question may sound stupid, this is because I'm not a shell32 expert.

Reply:
Thanks for the explanation

Reply:
Thanks for the explanation